### PR TITLE
Add common, utility scripts

### PR DESCRIPTION
**Pull request by st3v3nhunt**

### CODE REVIEW

File: scripts/start [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env sh
+
+set -e
+projectRoot="$(a="/$0"; a=${a%/*}; a=${a:-.}; a=${a#/}/; cd "$a/.." || return; pwd)"
+
+cd "$projectRoot"
+
+docker-compose down --volumes
+docker-compose up --build
+docker-compose down --volumes
```
Comment (johnwatson484) [inline scripts/start:10]: I don't see the value in this script.  If I want to start a service I can just run `docker-compose up --build` and stop it when I want, making a conscious decision to remove the volumes if I want to lose them.

With this approach we destroy all volumes, starts the service then immediately shuts it down destroying the volumes.  If I had data in those volumes I wanted to keep, I'd be frustrated that starting the service wiped all my data.

We also lose the capability to pass additional docker-compose arguments so I don't see why we'd use this and not just Docker Compose.

If a service has a complex start up, then it might warrant a `start` script and teams can make a decision to create their own.  But I'd rather see them optimising their Docker Compose setup so running a service can just use Docker Compose commands.

I don't think we should include this script in a minimal setup repo as the first thing I'd do is delete it.

File: scripts/test [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env sh
+
+set -e
+projectRoot="$(a="/$0"; a=${a%/*}; a=${a:-.}; a=${a#/}/; cd "$a/.." || return; pwd)"
+
+command="$*"
+project="ffc-template-node"
+service="ffc-template-node"
```
Comment (johnwatson484) [inline scripts/test:8]: Both of these variables have the same value and given the usage in the script, I don't see any value why a single value wouldn't suffice.

File: scripts/test [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  compose down --volumes
+
+  # Ensure container images are up to date
+  compose build
+
+  # Run tests
+  compose run "$service" "$command"
+
+  # Clean up
+  compose down --volumes
```
Comment (johnwatson484) [inline scripts/test:31]: This is something we have in our current demo repos and it's something about our setup I don't like.  It's not a problem with the script itself, but more how we use compose files and what will happen as a result of the script.

Using the [claim service compose file](https://github.com/DEFRA/ffc-demo-claim-service/blob/master/docker-compose.yaml), you'll see the postgres service is defined here so it is used by both local development and the tests.

That means they are sharing a volume, so if during my local development I add a load of test data to this service and then want to run the tests through this script, I will lose all of that data and have to keep adding it again between test runs which is going to frustrate developers.

A better way in my opinion is to have separate Postgres services defined in the `override` and `test` compose files, then this script won't affect my local development database as they have independent data volumes.  This is the approach I've followed elsewhere, example in this [override file](https://github.com/johnwatson484/dream-league-api/blob/master/docker-compose.override.yaml) and related [test file](https://github.com/johnwatson484/dream-league-api/blob/master/docker-compose.test.yaml)

If teams don't want to do that, then fine it's their call but this script is quite destructive if not.

File: scripts/test [hunk 4]
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env sh
```
Comment (johnwatson484) [inline scripts/test:1]: One of the frustrations I have with our current test scripts are how they work with file watching.  When I'm writing code, I tend to have the tests running continuously in the container using watch mode.  I'd like to offer a simple way for teams to do the same as I'm sure I'm not alone in running tests in watch mode.

With our current setup, the only real way to do this is to change the command in the `docker-compose.test.yaml` file and remember to not commit that change.  That's a bit of friction I'm not a fan of, especially as I seem to keep committing that change causing the CI to get stuck watch the files!

What I would like us to do is add a default `docker-compose.test.watch.yaml` file so we don't need to keep amending the commands.

In test scripts I've written elsewhere I have adapted test scripts such as this to use this file depending on whether a `-w` argument is passed.

[Here](https://github.com/johnwatson484/dream-league-api/blob/master/scripts/test) is an example of a test script I've written almost exactly like this but also allows file watching.

**Review by johnwatson484 — CHANGES_REQUESTED** [review]
Added my feedback regarding these scripts.

TL;DR I don't think we should include the start script in a minimum setup repo.  I like the test one, but think we should amend it and our Docker Compose files in general.

The test scripts gives me an opportunity to share my thoughts about our Docker Compose files so thanks for the opportunity!
